src/main/java/com/softdevsix/api/services/FileService.java

```diff
@@ -0,0 +1,80 @@
+package com.softdevsix.api.services;
+
+import com.softdevsix.api.domain.entities.file.File;
+import com.softdevsix.api.domain.entities.file.MethodCoverageResult;
+import com.softdevsix.api.repositories.IFileRepository;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+@Service
+public class FileService implements IFileService {
+    private IFileRepository fileRepository;
+
+    public FileService(@Qualifier("fileMemoryRepository") IFileRepository fileRepository) {
+        this.fileRepository = fileRepository;
+    }
+
+    @Override
+    public File getFileById(UUID fileId) {
+        return fileRepository.findById(fileId);
+    }
+
+    @Override
+    public float calculateFileCoverage(File file) {
+        float totalMethodCoverage = 0f;
+        int methodCount = 0;
+
+        for (MethodCoverageResult methodCoverage : file.getCoverageResult().getAllStatements()) {
+            int totalStatements = methodCoverage.getStatements().size();
+            int coveredStatements = (int) methodCoverage.getStatements().values().stream().filter(Boolean::booleanValue).count();
+
+            if (totalStatements > 0) {
+                totalMethodCoverage += (coveredStatements * 100f) / totalStatements;
+                methodCount++;
+            }
+        }
+
+        return methodCount > 0 ? totalMethodCoverage / methodCount : 0f;
+    }
+
+    @Override
+    public float calculateFileMethodCoverage(File file) {
+        int totalStatements = 0;
+        int coveredStatements = 0;
+
+        for (MethodCoverageResult methodCoverage : file.getCoverageResult().getAllStatements()) {
+            for (Map.Entry<Integer, Boolean> entry : methodCoverage.getStatements().entrySet()) {
+                totalStatements++;
+                if (entry.getValue()) {
+                    coveredStatements++;
+                }
+            }
+        }
+
+        return (totalStatements > 0) ? (coveredStatements * 100f) / totalStatements : 0f;
+    }
+
+    @Override
+    public List<Integer> getUncoveredLines(File file) {
```

Comment:
Consider receiving the fileId instead of the File object, to avoid data inconsistencies when the object in memory is out of date.